### DISCUSSION TITLE
What can i do if someone copies my code?

### DISCUSSION THREAD

**Discussion by yannic-md** [community/community#51766] [Programming Help]
I want to make my discord bots open source, but I'm pretty sure that many discord users will copy it if I will make it open source. So that comes to my question: What can I do if that happens? I mean, I can't see their bot code because they will probably don't upload it on github too.

**Comment by SmashedFrenzy16** [comment]
Hi @RazzerDE,

On GitHub, you can use licenses to preserve your code. Licenses help let you keep your rights over the code so that people have to credit you when they use your code. Different licenses allow different permissions for each piece of code your produce. This article should get you started quickly on licenses: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/licensing-a-repository

There are different licenses to choose from, some more restrictive than others. This website will help you pick the right license for you: https://choosealicense.com/

Hope this helps!

P.S. If you found this answer useful, could you please mark it as an answer? I am trying to get the Galaxy Brain achievement.

**Reply by priyazsh** [reply]
ok i choose a license but what if someone still copies my code without credits?

**Reply by yannic-md** [reply]
What if the discord users does still copies the code? That will happen I know that already. What can I do against it?

**Reply by SmashedFrenzy16** [reply]
@RazzerDE and @oyepriyansh they can't. If they do, you should then file a complaint to GitHub and they will sort out the issue.

**Reply by yannic-md** [reply]
How would that complaint look and what can GitHub do if someone uses my code for their own discord bot? Aside from that, the copied code would not be uploaded on github from the thief.
